warehouse/accounts/services.py

```diff
@@ -80,14 +82,19 @@ def __init__(self, session, *, ratelimiters=None, remote_addr, metrics):
         )
         self.remote_addr = remote_addr
         self._metrics = metrics
+        self.cached_get_user = functools.lru_cache()(self._get_user)
```

Comment:
Side effect of scrambling with figuring out why I couldn't get the latest `password_date` value... but interestingly this may resolve a memory leak? https://rednafi.github.io/reflections/dont-wrap-instance-methods-with-functoolslru_cache-decorator-in-python.html